lib/core/src/chain/bitcoin.rs

```diff
@@ -101,16 +120,20 @@ impl BitcoinChainService for ElectrumClient {
     }
 
     fn get_scripts_history(&self, scripts: &[&Script]) -> Result<Vec<Vec<History>>> {
-        let scripts: Vec<&bitcoin::Script> = scripts
-            .iter()
-            .map(|t| bitcoin::Script::from_bytes(t.as_bytes()))
-            .collect();
-
         Ok(self
             .client
-            .batch_script_get_history(&scripts)?
+            .batch_script_get_history(scripts)?
             .into_iter()
             .map(|e| e.into_iter().map(Into::into).collect())
             .collect())
     }
+
+    fn script_list_unspent(&self, script: &Script) -> Result<Vec<Unspent>> {
+        Ok(self
+            .client
+            .script_list_unspent(script)?
+            .into_iter()
```

Comment:
```suggestion
            .into_iter()
            .filter(|utxo| script.dust_value().lt(&bitcoin::amount::Amount::from_sat(utxo.value)))
```

Outputs below the dust limit should be ignored when deciding if a swap is refundable or not.

Reply:
Not sure about this @ok300. Also there can be several utxos which their aggregated amount is above dust.

Reply:
Agree. As discussed, we can leave this out.